src/Console/Command/ObfuscateData.php

```diff
@@ -0,0 +1,225 @@
+        $resultsCount = count($results);
+        foreach ($results as $row) {
+            $postEntity = $this->postRepository->getEntity($row);
+            $fakeEmail = $faker->safeEmail;
+            $fakeName = $faker->firstName." ".$faker->lastName;
+            $postEntity->setState([
+                'author_realname'=> $fakeName,
+                'author_email'=> $fakeEmail,
+            ]);
+            $overwrittenCount += $this->postRepository->update($postEntity);
+        }
+        if ($overwrittenCount == $resultsCount) {
+            $this->info("updated ".$overwrittenCount." records.\n");
+            return $overwrittenCount;
+        } else {
+            $this->info("failed to overwrite all records.\n");
+            return false;
+        }
+    }
+
+    private function overwriteSiteConfig()
+    {
+        $this->info("\nORemoving general settings data...");
+        $siteConfig = $this->configRepository->get('site');
+        $siteConfig->setState([
+            'name' => 'Test site',
+            'description' => 'Staging deployment for testing',
+            'email' => 'admin@ushahidi.com',
+            //TODO: reset these?
+            //'cient_url' => '??',
+            // 'url' => '??'
+        ]);
+        $this->configRepository->update($siteConfig);
+        $this->info("\nDone.");
+    }
+
+    private function removeDataProviderValues()
+    {
+        $this->info("\nORemoving dataProvider data...");
+        $dataProviderConfig = $this->configRepository->get('data-provider');
+        //TODO: walk through each key instead?
+        //TODO:  otherwise set them to defaults?
+        $dataProviderConfig->setState([
+            'url' => '',
+            'providers' => [
+                'smssync' => false,
+                'email' => false,
+                'twilio' => false,
+                'nexmo' => false,
+                'twitter' => false,
+                'frontlinesms' => false
+            ],
+            'frontlinesms'=> [],
+            'nexmo'=> [],
+            'smssync'=> [],
+            'twilio'=> [],
+            'email'=> [],
+        ]);
+        $this->configRepository->update($dataProviderConfig);
+        $this->info("\nSet all dataproviders to false.");
+    }
+
+    private function deleteUsers()
+    {
+        $this->info("Deleting users...");
+        $query = DB::delete('users');
+        $count = $query->execute($this->db);
+        $this->info("removed ".$count." records.\n");
+        return $count;
+    }
+
+    //TODO: unhardcode this if this is a feature we want
+    private function addAdminUser()
+    {
+        $newUserEntity = $this->userRepository->getEntity();
+        $newUserEntity->setState([
+            'email' => 'admin@ushahidi.com',
+            'realname' => 'Admin User',
+            'role' => 'admin',
+            'password' => 'admin' ]); // NOTE: pw is implicitly hashed via create
+        $id = $this->userRepository->create($newUserEntity);
```

Comment:
Just make sure to handle the state where admin@ushahidi.com already exists

Reply:
I'd assumed admin@ushahidi.com would have been wiped along with all other users, but I suppose we could just *not* wipe that account if it's there -- or ask in-line what the person running the script prefers.

Reply:
True. If we've already scrambled the email it won't be there. That said if we import say 10% of production deployments to staging and then all of them have a admin user with password admin... thats basically leaving the data open to the public. It might be easier just not to add the user since we have another console command that can do that if we need to.